www/src/pages/headless/combobox.md

```diff
@@ -0,0 +1,432 @@
+                }
+            }
+        }
+    }
+}
+```
+
+## Truncated result list
+
+The `results` Flow only displays a fixed number of items at a time, configured via the `maximumDisplayedItems` property.
+If more items match the given query than configured to be displayed, the `truncated` property available in the `results`
+Flow is set to `true` so an appropriate hint can be displayed.
+
+```kotlin
+comboboxItems {
+    results.render { (_, items, truncated) ->
+        items.forEach { item ->
+            comboboxItem(item) {
+                span { +item.value }
+            }
+        }
+        if (truncated) {
+            span {
+                +"Refine your query for more results"
+            }
+        }
+    }
+}
+```
+
+## Add a Label
+
+The combo box can be supplemented with a label using `comboboxLabel`, e.g. for use in forms or for a screen reader:
+
+```kotlin
+combobox<String> {
+    comboboxLabel {
+        span { +"Select a country" }
+    }
+}
+```
+
+## Open State
+
+Combobox is an [`OpenClose` component](#closable-content---openclose). There are different `Flow`s and `Handler`s
+like `opened` available in its scope to control the open state of the combo box based on state changes.
+
+Most of the time, the open state managed by the component itself should be enough for all use cases, though.
+
+## Transitions
+
+Showing and hiding the selection list can be easily animated with the help of `transition`:
+
+```kotlin
+comboboxItems {
+    transition(
+        opened,
+        enter = "transition duration-100 ease-out",
+        enterStart = "opacity-0 scale-95",
+        enterEnd = "opacity-100 scale-100",
+        leave = "transition duration-100 ease-in",
+        leaveStart = "opacity-100 scale-100",
+        leaveEnde = "opacity-0 scale-95"
+    )
+}
+```
+
+## Positioning
+
+`comboboxItems` is a [`PopUpPanel`](#floating-content---popuppanel) and therefore provides a set of
+configuration options to control the position or distance of the list box from the `listboxButton`
+as a reference element:
+
+```kotlin
+comboboxItems {
+    placement = PlacementValues.top
+    addMiddleware(offset(20))
+}
+```
+
+::: info
+In practice, the [`comboboxInput`](#comboboxinput) element might be wrapped in additional elements that are considered to be a part
+of it. Since by default the dropdown is positioned relative to the [`comboboxInput`](#comboboxinput) element, the dropdown may
+appear out of place. In those cases, the outermost wrapping element can be created via the
+[`comboboxPanelReference`](#comboboxpanelreference) brick to fix the positioning.
+:::
+
+The anchor element of the dropdown is determined based on a number of conditions:
+
+1) If a [`comboboxInput`](#comboboxinput) is present, the panel is placed relative to it.
+2) If a [`comboboxPanelReference`](#comboboxpanelreference) is present, it is _used as the reference instead_.
+3) Otherwise, the dropdown references a minimal fallback input element.
+
```

Comment:
I would suggest to add something like this:

"In order to see an example of the `comboboxPanelReference` in action, have a look at the source code of the combobox demo code within the `headless-demo` module. You can observe the consequences by removing and re-adding it there."